File: JavaScript/442-Find All Duplicates in an Array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDuplicates = function(nums) {
    let res = new Array();
    for (let i=0; i<nums.length; i++){
        let num = Math.abs(nums[i]);
        if (nums[num-1] < 0){
            res.push(num);
        }else{
            nums[num-1] *= -1;
        }
    }
    return res;
};

File: C++/442-Find All Duplicates in an Array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        vector<int> res;
        for (int num:nums) {
            if (nums[abs(num)-1] < 0) {
                res.push_back(abs(num));
            } else {
                nums[abs(num)-1] *= -1;
            }
        }
        return res;
    }
};